ddtrace/_monkey.py

```diff
@@ -211,6 +213,16 @@ def on_import(hook):
 
 
 def patch_all(**patch_modules):
+    deprecate(
+        "patch_all is deprecated and will be removed in a future version of the tracer.",
+        message="""patch_all is deprecated in favor of ``import ddtrace.auto`` and ``DD_PATCH_MODULES``
+        environment variable if needed.""",
+        category=DDTraceDeprecationWarning,
+    )
+    _patch_all(**patch_modules)
+
+
+def _patch_all(**patch_modules):
     # type: (bool) -> None
     """Enables ddtrace library instrumentation.
```

Comment:
We should move these docs to the public `patch_all(...)` function. We should also include a note saying this method is deprecated and `ddtrace.auto` should be used.